Tanks/Assets/Scripts/Tank/TankShooting.cs

```diff
@@ -68,21 +74,54 @@ private void Update()
             //we released the button, having not fired yet
             Fire();
         }
+        
+        // Check with weapon is currently selected
+
 
     }
 
 
     private void Fire()
     {
         // Instantiate and launch the shell.
+        
         m_Fired = true;
 
-        Rigidbody shellInstance = Instantiate(m_Shell, m_FireTransform.position, m_FireTransform.rotation) as Rigidbody;
-        shellInstance.velocity = m_CurrentLaunchForce * m_FireTransform.forward;
+        if (m_shellToFire == 1)
+        {
+            Rigidbody shellInstance = Instantiate(m_ShellOne, m_FireTransform.position, m_FireTransform.rotation) as Rigidbody;
+            shellInstance.velocity = m_CurrentLaunchForce * m_FireTransform.forward;
+
+            m_ShootingAudio.clip = m_FireClip;
+            m_ShootingAudio.Play();
+
+            m_CurrentLaunchForce = m_MinLaunchForce;
+            m_shellToFire++;
+        }
+        else if (m_shellToFire == 2)
+        {
+            Rigidbody shellInstance = Instantiate(m_ShellTwo, m_FireTransform.position, m_FireTransform.rotation) as Rigidbody;
+            shellInstance.velocity = m_CurrentLaunchForce * m_FireTransform.forward;
+
+            m_ShootingAudio.clip = m_FireClip;
+            m_ShootingAudio.Play();
+
+            m_CurrentLaunchForce = m_MinLaunchForce;
+            m_shellToFire++;
+        }
+        else if (m_shellToFire == 3)
+        {
+            Rigidbody shellInstance = Instantiate(m_ShellThree, m_MineTransform.position, m_MineTransform.rotation) as Rigidbody;
```

Comment:
Same as before, you don't have to use as Rigidbody.
The references are fine because you're actually using them here